src/main/java/dev/ebrydeu/spring_boot_library/domain/entities/Message.java

```diff
@@ -0,0 +1,30 @@
+package dev.ebrydeu.spring_boot_library.domain.entities;
+
+import jakarta.persistence.*;
+import lombok.*;
+import java.util.UUID;
+
+@Data
+@Entity
+@Table (name = "MESSAGE")
+
+@Getter
+@Setter
+@NoArgsConstructor
+public class Message {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @NonNull private UUID id;
+
+    @NonNull private String title;
+    @NonNull private String body;
+    @NonNull private String author;
+    @NonNull private String date; //can it be Date obj?
```

Comment:
1. `@NonNull` is a Lombok annotation that doesn't typically work for entities. Therefore, it should be removed. Additionally, it's important to clarify the intention behind its usage. If you're aiming to replicate a SQL constraint, then you need to use the `@Column` annotation and specify the constraint as `nullable`.
  
2. Regarding the date - indeed, you'll need to use any Date object along with adding the annotation `@CreationTimestamp`.


```suggestion
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(nullable = false)
    private UUID id;

    @Column(nullable = false)
    String title;
    @Column(nullable = false)
    String body;
    @Column(nullable = false)
    String author;

    @CreationTimestamp
    @Column(nullable = false)
    Instant date; // can also use LocalDateTime/Date/LocalDate/ZonedDateTime/Duration/OffsetDateTime/LocalTime

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id")
    private User user;


```